### ISSUE TITLE
Update the bot to be able to not require ssl connections.

### ISSUE THREAD

**Issue by sudosoldier** [rawsonj/triviabot#42]
~/triviabot$ ./trivia.py 
Save directory doesn't exist.
Lost connection ([Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
])

**Comment by rawsonj** [comment]
After looking at my (albeit old and crappy code) this isn't an error, it's a statement, that there is no save directory. It gets created after someone scores. Don't ask me what I was thinking when I wrote it, it's very old, and gone in my rewrite branch (if I ever get done with it).

**Comment by rawsonj** [comment]
So why did you file this?

**Comment by sudosoldier** [comment]
Well because I am actively using this bot on GameSurge

**Comment by rawsonj** [comment]
That's great, and I'm glad people are using it, but might I suggest you debug your network connection, since that is what your issue looks like to me.

**Comment by sudosoldier** [comment]
I was able to fix it a while ago. It was because he was trying to connect on SSL to a network using port 6667

**Comment by rawsonj** [comment]
And 6667 is commonly not used with ssl.

I suppose I should add configuration and a switch to use a different reactor method when not connecting via ssl. Currently, the bot is hard-coded to use ssl.

I will update this issue accordingly to reflect this and work on it when I have some time.

**Comment by sudosoldier** [comment]
It's pretty much my fault for not knowing python well. I just had to change the conf to whatever ssl the server I'm connecting to uses.

**Comment by rawsonj** [comment]
Well, and I could have been nicer to potential users by saying something in the readme like "THIS BOT REQUIRES TO CONNECT TO SSL PORTS."

Probably would have alleviated things.

So thanks for pointing this out, maybe some day I'll get to a point when I can add it in.

**Comment by beadon** [comment]
wow, thanks for this.  I was having the same problem.  Default config chooses 6667 as the port, should be 6697 as default SSL.

**Comment by andy5995** [comment]
This has now been added to [example_config.py](https://github.com/rawsonj/triviabot/blob/master/example_config.py)
`USE_SSL = "YES"`

And the appropriate code has been added to the main program.

I'll update the docs a little, and throw a comment in the config example about port settings.

> ~/triviabot$ ./trivia.py
Save directory doesn't exist.
Lost connection ([Failure instance: Traceback (failure with no frames): : Connection was closed cleanly.
])

This confused me the first time I saw it, too. I'll move the code up, so the directory is created the first time the program is run, regardless if it errors out for connection issues.
